pkg/build/controller/controller.go

```diff
@@ -190,6 +160,52 @@ func (bc *BuildController) nextBuildPhase(build *buildapi.Build) error {
 	return nil
 }
 
+// resolvePushspec determines the Docker pushspec where the build output image
+// should be pushed to, based on the value of build.Spec.Output.To, and sets
+// build.Status.Pushspec accordingly.
+func (bc *BuildController) resolvePushspec(build *buildapi.Build) error {
```

Comment:
since you're actually updating the build object, i'd call this setPushSpec or resolveAndSetPushSpec.  (also prefer PushSpec over Pushspec)